### PR TITLE
Add openlayers

### PR DESCRIPTION
**Pull request by Saela**
Add a simple map using open layers

### CODE REVIEW

**Comment by Tugark** [comment]
@Saela I pushed some changes:

* The map is now only shown once, by adding a return value in the useEffect() hook (since in development, `React.StrictMode` forces components to render twice, and OL initializes two maps which do not get removed once hooked)
* I switched from `useRef` to `useId` because I think that we only need a unique ID, and not the whole `ref` thing. If I am wrong, please undo it :)
* I found a nice [blogpost](https://medium.com/swlh/how-to-incorporate-openlayers-maps-into-react-65b411985744) that uses a contextwrapper so we can have the subcomponents of the mapwrapper nicely done as individual components, while allowing them to access the mapobject. As long as we're not manipulating state, that should be a fine approach.

**Comment by Tugark** [comment]
@Saela The current state works 🥳 Some things I did, and which you can maybe expand on for the create/edit functionality:

* I changed the coordinate system and added the Swisstopo WMS - because both Swisstopo as well as the OpenLayers WMS are provided in EPSG:3857, setting the view projection to EPSG:4326 led to a somewhat ugly distortion. We'll have to check whether we can get them in EPSG:4326. As of now, we're transforming our GeoJSON objects on the fly from 4326 to 3857 and have undistorted background maps 🪂 **However, this could be more difficult in the edit/create view. If it does not work, we'll have to use the `proj4` library to transform the coordinates on the fly.** I can show you this if need be (boy, how am I excited for working with geodate, and at the same time already triggered because of these projection issues 😆 )
* I added a function that generates our markers and a corresponding react component. This could certainly be improved to be more generic, because we're going to have more geometry types in the futures. However, it serves as a proof of concept and can also be used once we're adding the geotagged photos.

**Comment by Saela** [comment]
@Tugark We can now set markers and also get the coordinates back to save them to the database. This works for both the edit and create view. I also fixed the display of the map.

Create View: After setting two markers I remove the draw interaction. The user has then to move the already set markers to the position he wants if he made a mistake.

I re-used your marker layer but split it into creating the layer and adding the features. Please check if this is ok for you. :)

File: src/components/shared/map/layers/WayPointMarkerLayer.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        const mapConfigurationService = new MapConfigurationService()
+        const maxMarkerCount = mapConfigurationService.getMaxMarkerCount()
+        const startIcon = mapConfigurationService.getStartIcon()
+        const endIcon = mapConfigurationService.getEndIcon()
+
+        const setupMarkerLayer = (): VectorLayer<VectorSource> => {
+            let markerLayer = map!.getAllLayers().find((layer: Layer) => layer.getProperties()['name'] === 'marker_layer') as VectorLayer<VectorSource>
+            let layerExtent: number[]
+
+            if(!markerLayer) {
```
Comment (Tugark) [inline src/components/shared/map/layers/WayPointMarkerLayer.tsx:47]: Am I understanding that part correctly - we check whether the layer has been added yet, and if it has, we're adding the features to it and if it doesn't, we're creating the layer, right?

Under what circumstances can either of these two cases happen? And why do we need to call `marker:ayer.getSource()?.clear(true)`? I believe that would clear the whole layer and then immediately readd the points?

**Review by Tugark — COMMENTED** [review]
Awesome 🥳 Thank you!

Just one question that we might revisit in the future - we're passing the handle function now from the parent container and manipulate state in the parent container directly. I think we could try to bubble events from the children up to the parent (TourForm) or even use a state library. I faced the same issue as well and just opted for this way with the context, but let's revisit this later :D

**Pull request merged**